api/src/pcapi/core/reminders/repository.py

```diff
@@ -11,3 +11,15 @@ def get_user_ids_with_reminders(offer_id: int) -> list[int]:
     )
 
     return [reminder.userId for reminder in future_offer_reminders_query]
+
+
+def delete_reminders_on_offer(offer_id: int) -> None:
+    future_offer_reminders_query = (
+        db.session.query(FutureOfferReminder)
+        .join(FutureOfferReminder.futureOffer)
+        .filter(FutureOffer.offerId == offer_id)
+    )
+    future_offer_ids = [future_offer.futureOfferId for future_offer in future_offer_reminders_query]
+    db.session.query(FutureOfferReminder).filter(FutureOfferReminder.futureOfferId.in_(future_offer_ids)).delete(
+        synchronize_session=False
+    )
```

Comment:
Je suis pas sur de comprendre ici
Ce que je lis: 
```
future_offer_reminders_query = query
future_offer_ids = [id for result in query]
query(FutureOfferReminder).filter(future_offer_ids).delete()
```
Pourquoi on ne peut pas simplement faire  `future_offer_reminders_query.delete()` ?

Reply:
non sqlalchemy ne permet pas de delete si il y a une jointure mais tu as raison qu'il est possible de faire une requete plus optimale
```suggestion
def delete_reminders_on_offer(offer_id: int) -> None:
    future_offer_reminders_query = (
        db.session.query(FutureOfferReminder.id)
        .join(FutureOfferReminder.futureOffer)
        .filter(FutureOffer.offerId == offer_id)
    )
    db.session.query(FutureOfferReminder).filter(FutureOfferReminder.futureOfferId.in_(future_offer_reminders_query)).delete(
        synchronize_session=False
    )
```

Reply:
Je me suis posé la question de le faire comme ça, mais je ne pensais pas que ça fonctionnait avec le join, ça me va, je change

Reply:
le fait de faire une subquery permet d'éviter que le delete porte sur le join donc ça contourne bien le problème